src/electionguard/dlog.py

```diff
@@ -38,6 +40,18 @@ async def __discrete_log_internal(e: ElementModP) -> int:
     global __dlog_max_exp
     global __dlog_lock
 
+    if __dlog_lock is None:
+        # We cannot run asyncio.Lock() if we don't have an "event loop", which
```

Comment:
This is the description that should be on the PR, not in code. 

Perhaps add a bit more information on the method description.

Reply:
I guess we disagree about how comments can be useful to future readers of the code. This code is incredibly subtle, depending on details of how Python's lack of a multithreaded coding model works. In fact, if I really understand it, we could nuke all the asynchrony altogether, since Python never has two threads running concurrently. I'm a little hesitant to do that, so at least I can leave behind an artifact of my thinking for whoever tackles this next.

Reply:
i modified this. ill explain in a comment in my diff